watertap/unit_models/tests/test_stoichiometric_reactor.py

```diff
@@ -0,0 +1,698 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import pytest
+from pyomo.environ import (
+    ConcreteModel,
+    value,
+    check_optimal_termination,
+    units as pyunits,
+)
+
+from pyomo.util.check_units import assert_units_consistent
+from idaes.core import (
+    FlowsheetBlock,
+    UnitModelCostingBlock,
+)
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+)
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+)
+from idaes.core.util.testing import initialization_tester
+
+from watertap.property_models.multicomp_aq_sol_prop_pack import (
+    MCASParameterBlock,
+    ActivityCoefficientModel,
+    DensityCalculation,
+    MaterialFlowBasis,
+)
+
+from watertap.unit_models.stoichiometric_reactor import StoichiometricReactor
+from watertap.costing import WaterTAPCosting
+
+__author__ = "Timothy Bartholomew, Alexander Dudchenko"
+
+solver = get_solver()
+
+
+class TestStoichiometricReactor:
+    @pytest.fixture(scope="class")
+    def basic_unit_mass(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(dynamic=False)
+
+        default_dict = {
+            "solute_list": [
+                "Ca_2+",
+                "HCO3_-",
+                "Na_+",
+                "Cl_-",
+                "Mg_2+",
+            ],
+            "diffusivity_data": {
+                ("Liq", "Ca_2+"): 9.2e-10,
+                ("Liq", "HCO3_-"): 1.19e-9,
+                ("Liq", "Na_+"): 1.33e-9,
+                ("Liq", "Cl_-"): 2.03e-9,
+                ("Liq", "Mg_2+"): 0.706e-9,
+            },
+            "mw_data": {
+                "H2O": 18e-3,
+                "Ca_2+": 40e-3,
+                "HCO3_-": 61.0168e-3,
+                "Na_+": 23e-3,
+                "Cl_-": 35e-3,
+                "Mg_2+": 24.305e-3,
```

Comment:
Note- you shouldn't have to add mw_data manually anymore. These data should load automatically under the hood. Just FYI

Reply:
I will leave it as is for now.